kubernetes/affinity.md

```diff
@@ -0,0 +1,275 @@
+
+In fleet dictating where a container gets run is called [Affinity and Anti-Affinity][fleet-affinity]. In Kubernetes this is called the [nodeSelector][k8s-node-selector] and [nodeAffinity/podAffinity][k8s-node-affinity] fields under PodSpec.
+
+[comment]: # (nodeSelector and how it differs from fleet affinity/anti-affinity)
+
+## Affinity in fleet
+
+In the world of fleet, affinity and anti-affinity are achieved through [fleet-specific options in systemd unit-files][fleet-affinity]. These options include:
+
+* `MachineID`: Run a unit on a specific host.
+* `MachineOf`: Run a unit on the same host as another unit.
+* `MachineMetadata`: Run on a host matching some arbitrary metadata defined in `fleet.conf`.
+* `Conflicts`: Prevent a unit from running on the same node as some other unit.
+* `Global`: Runs a unit on every node.
+* `Replaces`: Run a unit in place of another unit.
+
+These are described [in greater depth at the fleet documentation][fleet-specific-options].
+
+These options allow one to specify with arbitrary granularly how, where, and when to schedule a unit on a fleet cluster.
+
+## Affinity in Kubernetes
+
+Kubernetes implements node affinity with the `nodeSelector` and `nodeAffinity` fields in PodSpec. These fields use both [pre-populated metadata][k8s-affinity-labels] or [user-defined metadata][k8s-user-node-labels].
+
+### User-defined metadata:
+
+These user-defined labels are set like so:
+
+```
+$ kubectl label nodes some-k8s-node.internal.hostename.ext key=value
+```
+
+And they can be retrieved like so:
+
+```
+$ kubectl get nodes --show-labels
+NAME                                       STATUS                     AGE       LABELS
+some-k8s-node.internal.hostname.ext        Ready                      8m        beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=instance-type,beta.kubernetes.io/os=linux,failure-domain.beta.kubernetes.io/region=some-us-region-2,failure-domain.beta.kubernetes.io/zone=some-us-region-2c,kubernetes.io/hostname=ip-10-0-0-101.some-us-region-2.internal,key=value
+...
+```
+
+### `nodeSelector`
+
+`nodeSelector` is the most basic way to set node affinity in Kubernetes. Given a set of `key: value` pair of requirements, a pod can be scheduled to run (or not run) on certain nodes.
+
+The general form of this node selection looks like this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  nodeSelector:
+    key1: value1
+    key2: value2
+```
+
+#### `nodeSelector` Example
+
+Let's say we're running a Kubernetes cluster on AWS and we want to run an Nginx pod on `m3.medium` instance. We also want a different `httpd` pod to only run in the `us-west-2` region and only on nodes with the `hello` key set to `world`. Here's how that `mypods.yaml` would look:
```

Comment:
Please make the selector like "example.com/load-balancer = true" or something more realistics.